app/thirdparty/java/other/writeily/widget/WrFilesWidgetFactory.java

```diff
@@ -51,40 +52,36 @@ public void onDataSetChanged() {
     }
 
     private void updateFiles() {
-        _widgetFilesList = (_dir == null) ? new File[0] : _dir.listFiles(file -> !file.isDirectory() && TextFormat.isTextFile(file));
-        if (_dir != null && _dir.equals(FilesystemViewerAdapter.VIRTUAL_STORAGE_RECENTS)) {
-            _widgetFilesList = FilesystemViewerCreator.strlistToArray(AppSettings.get().getRecentDocuments());
+        _widgetFilesList.clear();
+        final File dir = WrWidgetConfigure.getWidgetDirectory(_context, _appWidgetId);
```

Comment:
Cleaned up the logic in this class. Pulling the `dir` here rather than in constructor helps as sometimes the shared prefs file is not properly flushed (?) before we read it here. `dir` will now be updated correctly on subsequent calls to this function rather than being incorrect.